### PR TITLE
Improve equal operations.

### PR DESCRIPTION
**Pull request by zherczeg**
The code is simplified a lot (a lot of code is deleted).

### CODE REVIEW

**Comment by LaszloLango** [comment]
Please measure the PR on RPi2.

**Comment by zherczeg** [comment]
Since equal is not a frequent operation, the difference is not big. Although it slightly improved the perf in 12 cases.

Benchmark | Perf (sec) |
----: | ----: | 
3d-cube.js | 1.490 -> 1.491 : -0.068% | 
3d-raytrace.js | 1.539 -> 1.531 : +0.524% | 
access-binary-trees.js | 0.763 -> 0.754 : +1.205% | 
access-fannkuch.js | 3.253 -> 3.127 : +3.873% | 
access-nbody.js | 1.589 -> 1.561 : +1.736% | 
bitops-3bit-bits-in-byte.js | 0.793 -> 0.796 : -0.473% | 
bitops-bits-in-byte.js | 1.043 -> 1.045 : -0.279% | 
bitops-bitwise-and.js | 1.945 -> 1.907 : +1.959% | 
bitops-nsieve-bits.js | 2.559 -> 2.525 : +1.340% | 
controlflow-recursive.js | 0.548 -> 0.558 : -1.713% | 
crypto-aes.js | 1.480 -> 1.456 : +1.635% | 
crypto-md5.js | 0.929 -> 0.926 : +0.256% | 
crypto-sha1.js | 0.872 -> 0.876 : -0.445% | 
date-format-tofte.js | 1.173 -> 1.151 : +1.897% | 
date-format-xparb.js | 0.612 -> 0.594 : +2.904% | 
math-cordic.js | 2.000 -> 2.010 : -0.520% | 
math-partial-sums.js | 1.124 -> 1.106 : +1.535% | 
math-spectral-norm.js | 0.888 -> 0.897 : -0.966% | 
string-base64.js | 4.736 -> 4.691 : +0.945% | 
string-fasta.js | 2.267 -> 2.254 : +0.588% | 
Geometric mean: | +0.806% | 

Binary sizes (bytes)
old:167192
new:166872

However, the binary size is decreased by 300 bytes, which is useful.

File: jerry-core/ecma/operations/ecma-comparison.c [hunk 1]
```diff
@@ -38,41 +38,22 @@ ecma_value_t
...
+  if (x == y)
+  {
+    return ecma_make_simple_value (ECMA_SIMPLE_VALUE_TRUE);
+  }
 
-  if (is_types_equal)
+  if (ecma_are_values_integer_numbers (x, y))
   {
-    // 1.
+    // Note: the (x == y) comparison captures the true case.
```
Comment (LaszloLango) [inline jerry-core/ecma/operations/ecma-comparison.c]: Could you also fix comments in the touched functions? (`// -> /* */`)

**Comment by LaszloLango** [comment]
LGTM

File: jerry-core/vm/vm.c [hunk 2]
```diff
@@ -20,6 +20,7 @@
 #include "ecma-array-object.h"
 #include "ecma-builtins.h"
 #include "ecma-conversion.h"
+#include "ecma-comparison.h"
```
Comment (dbatyai) [inline jerry-core/vm/vm.c]: This should be above `ecma-conversion.h`

**Comment by dbatyai** [comment]
LGTM after the include is fixed

**Pull request merged**